bin/compile

```diff
@@ -1,28 +1,95 @@
 #!/bin/bash
 
+set -e
+
 BUILD_DIR=$1
-#CACHE_DIR=$2
+CACHE_DIR=$2
 ENV_DIR=$3
 BUILDPACK_DIR="$(dirname "$(dirname "$0")")"
 
+function main() {
+  export_env_dir "$ENV_DIR"
+
+  # Get latest stable version from the tailscale pkgs page
+  TAILSCALE_VERSION=${TAILSCALE_VERSION:-$(curl --silent https://pkgs.tailscale.com/stable/ | \
+    awk '/option/ {gsub("<[^>]+>", ""); print $1}' | \
+    grep -v latest | sort -r | head -1)}
```

Comment:
Wasn't familiar with this syntax. Apparently 

```
: ${TAILSCALE_VERSION:=${curl etc}}
```

is another way to do this?

Reply:
Interesting. This `:` operator is pretty curious. From [this stackoverflow](https://stackoverflow.com/questions/3224878/what-is-the-purpose-of-the-colon-gnu-bash-builtin) it appears as though it's equivalent to `true`. Which is a weird thing to consider for how it's being used in this example.

Personally, I sort of like duplicating the variable name like in my original construction. I think it makes it more clear what's happening (essentially, set the variable equal to the variable if it's already set, otherwise, the result of the curl command). That construction allows me to rename the variable if I want (which I did in test script).

I think I'm going to leave this one as-is if that's cool with you.

Reply:
Oh yeah, didn't mean to suggest a change. Just that I learned TWO things today.